### ISSUE TITLE
Workflow dispatch of deployment doesn't work

### ISSUE THREAD

**Issue by tobiasdiez** [JabRef/JabRefOnline#2205]
Reproduce: Create a new PR, go to actions, deploy workflow, select manual run based on the workflow.
Result: Fails at azure login because it cannot login. Reason is that azure federated credentials have to match exactly the subject registered in the azure portal, but we don't yet know which branch will be previewed. 

Blocked by https://github.com/Azure/login/issues/346.

Related:
- https://stackoverflow.com/questions/71051432/github-actions-azure-oidc-with-subject-value-for-any-branch
- Potential workaround: use https://learn.microsoft.com/en-us/cli/azure/identity/federated-credential?view=azure-cli-latest#az-identity-federated-credential-update to update the subject
